### DISCUSSION TITLE
Weird Issues with Adding Movies & Shows

### DISCUSSION THREAD

**Discussion by ngx3** [itsToggle/plex_debrid#266] [General]
hey, I have some weird issues with my plex_debrid. I add tv shows to my watchlist and when I add movies to "add to my watchlist" it just unchecks itself. They show up @ Available From Your Watchlist but I can't stream or watch anything, I gotta pay for it via Disney or w/e. I'm on Plex using Real Debrid.

**Comment by itsToggle** [comment]
Hi, the "available to stream on ..." fields on Plex Discover can be a bit slow to update. The movies/shows should show up inside your plex libraries. Movies are automatically removed from your watchlist.

**Reply by ngx3** [reply]
It's been an hour. I'm starting to doubt plex debrid is even running, how can I check> when I added my plex user token should I have put the plex server token from my tokens list or the web browser one is okay?

**Reply by itsToggle** [reply]
depends on your setup. are you running the "standard" version or the "dockerized version"? You would need to open the terminal that plex_debrid is running in, to make sure it's running

**Reply by itsToggle** [reply]
any of your device tokens are okay

**Reply by ngx3** [reply]
its the standard. do I have to keep it open in the terminal on the main menu?

**Reply by itsToggle** [reply]
No. As you can read in the readme, you need to choose "1) Run" to run the download automation.

**Reply by itsToggle** [reply]
You need to keep the terminal open in order for the script to be able to run.

**Reply by ngx3** [reply]
Got it. I just noticed. anyway to run it as a background service or something without me needing to keep this terminal open? not sure if i can add the --daemon as when i installed the service.
> Type 'exit' to return to the main menu.
> [plex] getting entire plex library ... done
> [plex error]: Your library seems empty. To prevent unwanted behaviour, no further downloads will be started. If your library really is 
> empty, please add at least one media item manually.

**Reply by itsToggle** [reply]
there are plenty of options to run python scripts in the background. Google is your friend, but you can also check out the discord channel.

The error message you see can be solved by adding at least one torrent to your debrid service, so it shows up on your rclone mount and therefore inside your Plex library.

**Reply by ngx3** [reply]
guess thats why it wasn't downloading then. thanks a lot for the help and for making this script.

**Reply by ngx3** [reply]
Sorry, quick question I downloaded a movie from a torrent using real debrid but it wasn't added to my Plex and my library isnt updating. Also the tv show I added to my watchlist isnt updating in the main menu of the script

**Reply by itsToggle** [reply]
did you create plex libraries of the mounted virtual drive? Do you still have rclone running? Do you see the torrents when you access the mounted virtual drive through a file explorer? 
What do you mean by "updating in the main menu"?

**Reply by ngx3** [reply]
yeah, I went through the script and added the libraries to the mounted drive and when I set up Plex, I chose the virtual drive folder I created with rclone. nope I don't think rclone is running. Is there a way to make it run again and stay running, I am using your version of rclone.


> "updating in the main menu"?

When I run the script, nothing runs, like in this image: [https://user-images.githubusercontent.com/71379623/185643627-45217303-75d8-4c9d-8c8b-41bb2e27fd87.gif](status)

**Reply by itsToggle** [reply]
the gif isnt loading for me. 
You need to either keep the rclone terminal open, or add the tag "--daemon" (linux) or "--no-console" (windows). This is all detailed in the readme. For your first run, its probaply best to keep the terminal open so you can look out for any errors.

**Reply by ngx3** [reply]
https://user-images.githubusercontent.com/71379623/185643627-45217303-75d8-4c9d-8c8b-41bb2e27fd87.gif

I tried running the ./rclone-linux command again with `sudo ./rclone-linux mount realdebrid: /opt/ng/plex_storage: --dir-cache-time 10s --daemon` and it outputs Fatal error: mount not ready. I searched it up and it says its is a broken package build and to contact the maintainer. I'm using your fork of rclone btw.

**Reply by itsToggle** [reply]
The package is not broken. It's working for me and virtually anyone else that's using it. Run it without "--daemon" for now and leave the terminal open. 

To run plex_debrid, you need to select option "1) run" from the main menu to start the download automation. Leaving the script in the main menu does not start the download automation.

What exactly is your issue? Please be more specific in what you are doing, what you are expecting, and what actually happening. 

Please tell me if:
1) you have rclone running
2) you can see the 3 folders "movies", "shows", and "default" in your mount location
3) there are no errors in the rclone terminal window
4) you have plex media server running
5) you have created a plex '"movie" library inside plex media server, for which you have chosen the mounted "movies" folder
6) you have created a plex "shows" library inside plex media server, for which you have chosen the mounted "shows" folder
7) you have finished the initial setup for plex_debrid
8) you have chosen option "1) run" to run the download automation 
9) there are no errors inside the plex_debrid terminal.

**Reply by I-am-PUID-0** [reply]
> https://user-images.githubusercontent.com/71379623/185643627-45217303-75d8-4c9d-8c8b-41bb2e27fd87.gif
> 
> I tried running the ./rclone-linux command again with `sudo ./rclone-linux mount realdebrid: /opt/ng/plex_storage: --dir-cache-time 10s --daemon` and it outputs Fatal error: mount not ready. I searched it up and it says its is a broken package build and to contact the maintainer. I'm using your fork of rclone btw.
> 

This typically occurs when the directory for the mount -- /opt/ng/plex_storage: -- doesn't exist.    

Try removing the colon at the end of /opt/ng/plex_storage:

e.g., sudo ./rclone-linux mount realdebrid: /opt/ng/plex_storage/ --dir-cache-time 10s --daemon

**Reply by ngx3** [reply]
@sonofmp3 It didn't work. I get the same error.

**Reply by itsToggle** [reply]
Try running it without --daemon, and read my last message

**Reply by ngx3** [reply]
Okay! I got it running, it was a fuse issue and a kernel module that was missing... even though it ran the first time...

**Rclone running the background (prompt)**: NOTICE: realdebrid root '': --vfs-cache-mode writes or full is recommended for this remote as it can't stream

**plex debrid running in the background (prompt)**:  Type 'exit' to return to the main menu.
[plex] getting entire plex library ... done
[plex error]: Your library seems empty. To prevent unwanted behaviour, no further downloads will be started. If your library really is empty, please add at least one media item manually.

I manually downloaded a torrent on real debrid and nothing happened. I tried adding a movie to my watchlist and nothing happened. I checked both screens and no errors reported on rclone or plex debrid. Not sure what to do now

**Reply by itsToggle** [reply]
Okay now that rclone is running we can focus on why plex_debrid is not downloading anything. The error message you are seeing in your plex_debrid console is telling you that either the server is not reachable, or that your Plex libraries are empty.

Please make sure that:
1) Plex is running
2) You've created 2 libraries in Plex, one for tv shows and one for movies
3) You've linked these libraries to your mounted folders (Add the "shows" and the "default" folder to your Plex tv show library, and add the "movies" and the "default" folder to your Plex movies library)
4) You can see and stream the movies/shows inside plex

If you've successfully done all of the above, we can check the server address you've entered into plex_debrid. You should add the address in the format `https://localhost:32400`

**Reply by ngx3** [reply]
![plex 1](https://user-images.githubusercontent.com/16364958/204092391-34e72a93-e95f-4b47-9c35-88cfd2d93b54.png)

1. Plex is running
2. Both Shows and Movies are linked to my /storage virtual drive, created with the mount
3. I did that when i set up plex via localhost:32400
4. i can see the folders inside my server

localhost:32400 is set when i ran plex debrid. in the photo I have Movies and Shows then I have Plex's movies and shows, so when i go to add a movie to my watchlist i add it then go to home, but my home is different than the one on the sidebar listed movies & shows on plex.

**Reply by itsToggle** [reply]
So your "movies" plex library and your "shows" plex library are not empty? Could you tell me the exact server address that youve entered into plex_debrid?

**Reply by ngx3** [reply]
No, both libraries are empty. I set http://localhost:32400 now. I had put http://myserverip:32400 before. I changed the address as an attempt to fix this issue.

**Reply by itsToggle** [reply]
They should not be empty if you have torrents on realdebrid. Are you running this on linux? If so, as stated in the readme, make sure to add the flag "--allow-other" to your rclone mount command, so that plex can see your mounted files.

As a side note, The error message is clearly telling you that either the plex libraries are empty or that the server is not reachable. If you know your plex libraries are empty, this is obviously the cause for the error message. Why wouldnt you mention this? In the future please read the readme and describe your problem more clearly so that I dont have to poke around forever to find your issue.

**Comment by ngx3** [comment]
> They should not be empty if you have torrents on realdebrid. Are you running this on linux? If so, as stated in the readme, make sure to add the flag "--allow-other" to your rclone mount command, so that plex can see your mounted files.

im on linux and yes, i have done that. I read the readme file. My issue is why isn't the script or rclone picking up the movies from Real debrid. ive scanned my library files and nothing shows up.

The virtual drive's contents:

```
drwxr-xr-x 1 root root    0 Jan  2  2000 default/
drwxr-xr-x 1 root root    0 Jan  2  2000 movies/
drwxr-xr-x 1 root root    0 Jan  2  2000 shows/
```

**I do see the movies from Real debrid in the movies folder, but nothing shows up on Plex's library.**

 The error doesn't state if the server is unreachble, ive also tested this with streamio and my debrid api, totally works, finds the movie files on there.

**Reply by itsToggle** [reply]
make sure the Plex user owns the mounted directory. Again, this has nothing to with either plex_debrid or rclone, this is just Linux stuff you can easily find googling "Plex doesn't find files on linux". The command would be `chown -R plex:plex your/mount/location`.

**Reply by ngx3** [reply]
It's alright, --allow-other doesn't seem to work, and you cannot change the owner of the folder of a mount. I'll post back with a solution for others.

**Reply by itsToggle** [reply]
You could also change the group the Plex user is in. You could place it into the root group, so it definitely has access.

**Reply by shivamgrover21** [reply]
Hey mate, i'm stuck here as well, What was the solution for you?
